Userland/Applications/TextEditor/MainWidget.cpp

```diff
@@ -660,7 +670,15 @@ bool MainWidget::open_file(const String& path)
 
     m_editor->set_text(file->read_all());
 
-    set_path(LexicalPath(path));
+    if (!path.starts_with("/"sv)) {
+        char* buf = getcwd(nullptr, 0);
+        LexicalPath lexical_path = LexicalPath::join(String(buf), path);
```

Comment:
I think you don't need the explicit `String(buf)` here, as the `String(char const*)` is not marked `explicit`. You can just do:
```cpp
LexicalPath::join(buf, path);
```
Also, you could use `auto lexical_path` to avoid repeating yourself.